trlx/trainer/accelerate_base_trainer.py

```diff
@@ -468,18 +475,40 @@ def learn(self):  # noqa: C901
         for _ in range(self.config.train.epochs):
             # For each batch
             for batch in self.train_dataloader:
+                mbs = [
+                    PPORLBatch(
+                        query_tensors=batch.query_tensors[mbi * self.mb_size : (mbi + 1) * self.mb_size],
+                        response_tensors=batch.response_tensors[mbi * self.mb_size : (mbi + 1) * self.mb_size],
+                        logprobs=batch.logprobs[mbi * self.mb_size : (mbi + 1) * self.mb_size],
+                        values=batch.values[mbi * self.mb_size : (mbi + 1) * self.mb_size],
+                        rewards=batch.rewards[mbi * self.mb_size : (mbi + 1) * self.mb_size],
+                    )
+                    for mbi in range(self.num_mb)
+                ]
                 # For each update per batch
                 for _ in range(self.n_updates_per_batch):
                     # Note that whereas standard policy gradient methods perform one
                     # gradient update per batch, PPO for example commonly performs
                     # multiple gradient updates on the same batch of data.
                     # https://arxiv.org/pdf/1707.06347.pdf
-                    forward_time = time()
-                    loss, stats = self.loss(batch)
-                    forward_time = time() - forward_time
-                    backward_time = time()
-                    self.accelerator.backward(loss)
-                    backward_time = time() - backward_time
+                    forward_time = 0
+                    backward_time = 0
+                    stats_accum = []
+                    for mb in mbs:
```

Comment:
to avoid unnecessary gradient synchronization when doing using gradient accumulation you can simply add: `self.accelerator.accumulate(self.model)` here